service/awsconfig/v10/ebs/ebs.go

```diff
@@ -86,24 +86,41 @@ func (e *EBS) DeleteVolume(ctx context.Context, volumeID string) error {
 
 // DetachVolume detaches an EBS volume. If force is specified data loss may occur. If shutdown is
 // specified the instance will be stopped first.
-func (e *EBS) DetachVolume(ctx context.Context, volumeID string, attachment VolumeAttachment, force bool, shutdown bool) error {
-	e.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("detaching EBS volume %s from instance %s", volumeID, attachment.InstanceID))
-
+func (e *EBS) DetachVolume(ctx context.Context, volumeID string, attachment VolumeAttachment, force bool, shutdown bool, wait bool) error {
 	if shutdown {
 		e.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("stopping instance %s", attachment.InstanceID))
 
-		_, err := e.client.StopInstances(&ec2.StopInstancesInput{
+		i := &ec2.StopInstancesInput{
 			InstanceIds: []*string{
 				aws.String(attachment.InstanceID),
 			},
-		})
+		}
+		_, err := e.client.StopInstances(i)
 		if err != nil {
 			return microerror.Mask(err)
 		}
 
+		if wait {
```

Comment:
Putting the wait block here. In practise waiting for the instance to be stopped took like less than 30 seconds. Would actually be cool to decouple this from blocking the resource but this 30 seconds should be ok now. We can optimize later.

Reply:
That timing matches my experience. Maybe if the master is under heavy load might take longer. Let's see and not optimize too early.